conda_build/source.py

```diff
@@ -66,16 +69,15 @@ def download_to_cache(cache_folder, recipe_path, source_dict, verbose=False):
         source_dict["fn"] if "fn" in source_dict else basename(source_urls[0])
     )
     hash_added = False
-    for hash_type in ("md5", "sha1", "sha256"):
-        if hash_type in source_dict:
-            if source_dict[hash_type] in (None, ""):
-                raise ValueError(f"Empty {hash_type} hash provided for {fn}")
-            fn = append_hash_to_fn(fn, source_dict[hash_type])
-            hash_added = True
-            break
+
+    for hash_type in set(source_dict).intersection(ACCEPTED_HASH_TYPES):
```

Comment:
If there's more than one hash key in the source_dict, this will randomize their order, so the `fn` assignment below will change every now and then. I think we should revert to the `for` loop +  break solution we had before.

Reply:
```suggestion
    for hash_type in sorted(set(source_dict).intersection(ACCEPTED_HASH_TYPES)):
```